lec15-data-viz.Rmd

```diff
@@ -1,4 +1,341 @@
+
+
+Let's say we don't want to facet, but we want to colour the dots depending on the species:
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()
+```
+
+We can change other aesthetic features of the graph. Let's get rid of the color aspect of our structure and change the colour and size of the points:
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length)) + geom_point(color="steelblue", size=4)
+
+```
+
+Going back to our graph coloured by species, what if we want to change other aspects, like the theme, or centre the title?
+
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+  
+```
+What about changing the font? Let's get rid of the theme for a moment, which changes many aspects of the aesthetics at once. Let's spend some time building our own theme...
+
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme(plot.title = element_text(hjust = 0.5), 
+        text = element_text(size = 8, family='Times'),
+        legend.title=element_blank())
+  
+```
+
+
+You can add your own alterations to themes. For example, here is an APA theme built on top of the base theme_bw(). Themes can be saved into variables so they don't need to be typed out every time you use a graph.
+
+
+```{r}
+
+apatheme <- theme_bw()+
+  theme(panel.grid.major=element_blank(),
+        panel.grid.minor=element_blank(),
+        panel.border=element_blank(),
+        axis.line=element_line(),
+        text=element_text(family='Times'),
+        legend.title=element_blank())
+
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  apatheme
+  
+```
+
+
+## Bar graphs
+
+What if we want a bar graph of Sepal Length by Species?
+```{r}
+ggplot(iris, aes(x=Species, y=Sepal.Length)) + geom_bar(stat="identity")+
+  ylab("Sepal Length") +
+  xlab ("Species") +
+ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+```
+
+## Violin plots
+
+```{r}
+ggplot(iris, aes(x=Species, y=Sepal.Length)) + geom_violin()+
+  ylab("Sepal Length") +
+  xlab ("Species") +
+ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+```
+
+## Box plots
+
+
+ We can use boxplots to visualize the distribution Sepal Length within each species:
+
+ ```{r boxplot}
+ ggplot(data = iris, aes(x = Species, y = Sepal.Length)) +
+     geom_boxplot()
+ ```
```

Comment:
Remove the leading space from all the lines in this code chunk

Reply:
addressed